src/backends/perf_pt/collect.c

```diff
@@ -144,16 +144,21 @@ bool perf_pt_free_tracer(struct tracer_ctx *tr_ctx);
 /*
  * Read data out of the AUX buffer.
  *
- * Reads `size` bytes from `aux_buf` into `trace_buf`, updating `*trace_len`.
+ * Reads from `aux_buf` (whose meta-data is in `hdr`) into `trace`.
  *
  * Returns true on success and false otherwise.
  */
 static bool
-read_aux(void *aux_buf, __u64 size, __u64 head_monotonic, __u64 *tail_p,
-         struct perf_pt_trace *trace)
+read_aux(void *aux_buf, struct perf_event_mmap_page *hdr, struct perf_pt_trace *trace)
 {
+    // atomic loads not strictly needed on x86_64, but we play it by the book.
```

Comment:
It's worth noting that the atomic reads are doing two things: they have a run-time effect (but, I agree, on x86_64, atomic/non-atomic reads are the same at run-time); but they have a compile-time effect (they prevent the compiler moving things around). So I don't think this line is true: without using atomics, it's possible the compiler could change the semantics of our program.

Reply:
It's true that these ordered atomic ops impose compiler barriers. Let me revise the comment.

Reply:
I think it can just be deleted...